test/unit/math/prim/mat/fun/dot_product_test.cpp

```diff
@@ -1,2 +1,2 @@
-#include <stan/math/prim/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/mat.hpp>
```

Comment:
Same as for the dot_product.hpp I guess. Please revert this change back if it has no other effect.